crates/rpc/rpc/src/eth/pubsub.rs

```diff
@@ -372,7 +380,46 @@ where
             futures::stream::iter(headers)
         })
     }
+}
+
+impl<N: NodePrimitives<SignedTx: TxHashRef>, Eth> EthPubSubInner<Eth>
+where
+    Eth: RpcNodeCore<Provider: CanonStateSubscriptions<Primitives = N>>
+        + EthApiTypes<
+            RpcConvert: RpcConvert<Primitives = N, Network = Eth::NetworkTypes, Error = Eth::Error>,
+        >,
+{
+    /// Returns a stream that yields all transaction receipts from new blocks.
+    fn transaction_receipts_stream(
+        &self,
+    ) -> impl Stream<Item = reth_rpc_convert::RpcReceipt<Eth::NetworkTypes>> {
+        let eth_api = self.eth_api.clone();
+        self.eth_api.provider().canonical_state_stream().flat_map(move |new_chain| {
+            let mut all_receipts = Vec::new();
+
+            // Process all blocks in the committed chain - exactly like new_headers_stream pattern
+            for (block, receipts) in new_chain.committed().blocks_and_receipts() {
+                // Build ConvertReceiptInput using our helper function
+                let inputs = build_convert_receipt_inputs(block, receipts);
```

Comment:
then we perhaps might not need this because we need to filter them receipts based on the hashes input